expression/builtin_cast.go

```diff
@@ -1674,9 +1674,13 @@ func WrapWithCastAsString(ctx sessionctx.Context, expr Expression) Expression {
 	if expr.GetType().EvalType() == types.ETString {
 		return expr
 	}
+	argLen := expr.GetType().Flen
+	if expr.GetType().EvalType() == types.ETInt {
```

Comment:
``` go
I think we'd better not use types.UnspecifiedLength in this case, 
since we know the max length of every integer type. 
We can use the max length of individual integer type as the argLen.

refer: https://dev.mysql.com/doc/refman/8.0/en/integer-types.html
```

Reply:
done, PTAL